source/DmapSupport.cpp

```diff
@@ -372,19 +395,79 @@ HRESULT SendIOControlCodeToController(HANDLE & handle, IOControlCode^ iOControlC
     }
 
     auto workItem = ref new WorkItemHandler(
-        [device, iOControlCode, ioControlCompleted, &hr]
+        [device, iOControlCode, ioControlCompleted, &hr, &bufferToDriver, &bufferFromDriver]
     (IAsyncAction^ workItem)
     {
-        create_task(device->SendIOControlAsync(iOControlCode, nullptr, nullptr)).then([&ioControlCompleted, &hr](UINT result)
+        create_task(device->SendIOControlAsync(iOControlCode, bufferToDriver, bufferFromDriver)).then([&hr, &ioControlCompleted](task<unsigned int> t)
```

Comment:
Why is `iOControlCompleted` passed a reference here, but as a copy above? If it is a HANDLE, then it's already a pointer right? Also, if we are expecting the inner-most lambda to alter it's value, then we would want to pass that all the way back out, and therefore it should be passed by reference into the preceding lambda, correct?